ocaml/xapi/xapi_vif_helpers.ml

```diff
@@ -294,6 +294,30 @@ let create ~__context ~device ~network ~vM ~mAC ~mTU ~other_config
       if List.mem device all_devices then
         raise
           (Api_errors.Server_error (Api_errors.device_already_exists, [device])) ;
+
+      (* If the VM uses a PVS_proxy, then the proxy _must_ be associated with
+         the VIF that has the lowest device number. Check that the new VIF
+         respects this. *)
+      ( match all_devices with
+      | [] ->
+          ()
+      | hd :: tl ->
+          let min_device = List.fold_left min hd tl in
+          let vm_has_pvs_proxy =
+            List.exists
+              (fun vif ->
+                Pvs_proxy_control.find_proxy_for_vif ~__context ~vif <> None
+              )
+              all
+          in
```

Comment:
Is it not enough to check that just the min_device's VIF has a PVS proxy - since otherwise this code wouldn't let a VIF get created? Perhaps the fold_left above would return `vif * vif_device` instead?

Reply:
You're right, I can simplify this.

Reply:
Also, this compares strings rather than integers, so have to fix that.

Reply:
Fixed that now.